### PR TITLE
Fix creating colormap from list of Color objects

### LINKED ISSUE #1858: Initialize ColorArray from list of color fail
When initialising `CollorArray` from list of `Colors` it fails. It works if on list there is at least one definition color with string.

This works:
```python
array1 = ColorArray(["b", "w"])
array2 = ColorArray(["black", "white"])
array3 = ColorArray(["black", Color((1, 1, 1))])
```

This fails:
```python
array4 = ColorArray([Color((0,0,0)), Color((1, 1, 1))])
```
with:
```
TypeError: float() argument must be a string or a number, not 'ColorArray'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/czaki/Projekty/partseg/vispy_test.py", line 6, in <module>
    array4 = ColorArray([Color((0,0,0)), Color((1, 1, 1))])
  File "/home/czaki/.pyenv/versions/partseg3.6-pyside/lib/python3.6/site-packages/vispy/color/color_array.py", line 148, in __init__
    rgba = _user_to_rgba(color, clip=clip)
  File "/home/czaki/.pyenv/versions/partseg3.6-pyside/lib/python3.6/site-packages/vispy/color/color_array.py", line 55, in _user_to_rgba
    color = np.atleast_2d(color).astype(np.float32)
ValueError: setting an array element with a sequence.
```

### PR DESCRIPTION
**Pull request by Czaki**
fix #1858 with adding check if any list member is instance of `Color` or `ColorArray`

### CODE REVIEW

File: vispy/color/color_array.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def _user_to_rgba(color, expand=True, clip=False):
         color = color.rgba
     # We have to treat this specially
     elif isinstance(color, (list, tuple)):
-        if any(isinstance(c, string_types) for c in color):
+        if any(isinstance(c, (string_types, Color, ColorArray)) for c in color):
```
Comment (kmuehlbauer) [inline vispy/color/color_array.py:50]: `Color` is not needed here, since it is a subclass of `ColorArray`.

File: vispy/color/tests/test_color.py [hunk 2]
```diff
@@ -344,4 +344,9 @@ def test_normalize():
     assert_allclose([y.min(), y.max()], [0.2975, 1-0.2975], 1e-1, 1e-1)
 
 
+def test_colormap_creation():
```
Comment (kmuehlbauer) [inline vispy/color/tests/test_color.py:347]: If you are already here, do you mind adding some more tests for the different parametrizations?

**Comment by Czaki** [comment]
I do not understand travis fail.

**Comment by djhoese** [comment]
Looks like a pytest-sugar incompatibility with newer pytest: https://github.com/conda-forge/pytest-sugar-feedstock/pull/9 and has been released as pytest-sugar 0.9.3 but isn't available on conda-forge yet: https://github.com/conda-forge/pytest-sugar-feedstock/pull/9

I've commented there to have it merged. After it is merged, we wait an hour, then I'll restart the tests.

**Comment by djhoese** [comment]
Ugh looks like it is broken with 0.9.3 too: https://github.com/pytest-dev/pytest/issues/3170#issuecomment-624504772

You could try limiting the version of pytest-sugar in the travis configs or just wait until things are released.

**Comment by Czaki** [comment]
> Ugh looks like it is broken with 0.9.3 too: [pytest-dev/pytest#3170 (comment)](https://github.com/pytest-dev/pytest/issues/3170#issuecomment-624504772)
> 
> You could try limiting the version of pytest-sugar in the travis configs or just wait until things are released.

I change sugar version but there is error:
```
RuntimeError: Could not import any of the backends. You need to install any of ['PyQt4', 'PyQt5', 'PySide', 'PySide2', 'Pyglet', 'Glfw', 'SDL2', 'wx', 'EGL', 'osmesa']. We recommend PyQt
```

**Comment by djhoese** [comment]
That's actually expected error for that set of tests I think. And sorry I was wrong, the problem wasn't necessarily pytest-sugar but its incompatibility with the new version of pytest. I think `pytest` needs to be limited not pytest-sugar.

**Comment by Czaki** [comment]
So any changes are requested? Which version of pytest is ok?

**Comment by djhoese** [comment]
`pytest` less than `5.4` should work (based on the bug reports I've seen).

**Comment by Czaki** [comment]
Fixed

**Comment by djhoese** [comment]
@kmuehlbauer Are you ok with this then? Feel free to merge if you are.

**Comment by kmuehlbauer** [comment]
@Czaki 

Thanks again for this fix.

I do not want to force this upon you. Could you by chance make two commits out of this four, one with your fixes and one with the travis changes? It would be easier to reverse the travis-changes, when fixed upstream. Normally we would have squash-merged.

If you are unsure I can do it, keeping your authoring information intact. Please let me know.

**Comment by Czaki** [comment]
Done

**Comment by kmuehlbauer** [comment]
@Czaki I'll merge as soon, the CI's are happy.

@djhoese I was about to add the `automerge` label... :grin:

**Comment by kmuehlbauer** [comment]
@Czaki Thanks again!

**Pull request merged**